src/components/search/index.js

```diff
@@ -72,9 +108,135 @@ const PoweredBy = () => (
   </div>
 )
 
+//
+//
+//
+
 const indices = [{ name: `Pages`, title: `Pages`, hitComp: `PageHit` }]
 
-export default function Search({ collapse }) {
+function SearchContent({
+  searchQuery,
+  focus,
+  onFocus,
+  onRemoveFocus,
+  collapse,
+}) {
+  const NO_LINE_SELECTED = -1
+  // By default we use -1 and not 0 to avoid displaying a selected background on the first result of
+  // the list, which can be weird when using the mouse to hover an item of the list: it would display
+  // the first line and the hovered line with the same background.
+  // This way we enter the "keyboard navigation mode" only when up/down is used.
+  const [selectedIndex, setSelectedIndex] = useState(NO_LINE_SELECTED)
+
+  React.useEffect(() => {
+    // reset selected index if we reset the search query
+    if (searchQuery?.length === 0) {
+      setSelectedIndex(NO_LINE_SELECTED)
+    }
+  }, [searchQuery, setSelectedIndex])
+
+  //
+  // keyboard shortcuts handling
+  //
+
+  const handleKeyDown = e => {
+    // we use javascript to retrieve the number of hits.
+    // tried to connect this SearchContent component to retrieve the number of hits using
+    // connectHits, connectStats, and connectStateResults on Search but it didn't worked.
+    const nbHits = document.querySelectorAll(".hits-wrapper .hit")?.length
+
+    // arrow up/down button should select next/previous list element
+    if (e.keyCode === 38) {
```

Comment:
could we use constants `e.keyCode === KEYS.ARROW_UP` or something, it's pretty confusing to have to look up constants

Reply:
should we use a constant Keys for the whole app in an util or define constants on this file only?